runtime/test/buffer.cc

```diff
@@ -267,11 +267,11 @@ TEST(buffer, for_each_contiguous_slice) {
   buf.allocate();
   int slices = 0;
   for_each_contiguous_slice(buf, [&](index_t slice_extent, char* slice) {
-    memset(slice, 7, slice_extent);
+    std::fill_n(slice, slice_extent, 7);
     slices++;
   });
   ASSERT_EQ(slices, 1);
-  ASSERT_TRUE(is_filled_buffer(buf, 7));
+  ASSERT_THAT(flatten(buf), testing::Contains(7).Times(buf.elem_count()));
```

Comment:
If this is the way we want to go, fine, but IMHO this just adds extra complexity/verbosity that doesn't really pay for itself. I'd vote instead for using (say) QCHECK_EQ().

Reply:
I'm also trying to get rid of for_each_index, which is also complex. The main reason I don't like complexity here is it's possible for there to be bugs in these checks, not just the code they are testing. At least the ASSERT_THAT matchers are less likely to have bugs than our own code.

That said, maybe it's a lateral move. I'll move the bug fix to a different branch and abandon this PR for now.